File: Benchmarks/enterprise/java/jakarta/testcode/BenchmarkTest00257.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import jakarta.ws.rs.*;
import jakarta.ws.rs.core.Context;
import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;

@Path("/")
public class BenchmarkTest00257 {

    @GET
    @Path("/BenchmarkTest00257")
    @Produces(MediaType.APPLICATION_JSON)
    public Response BenchmarkTest00257(@QueryParam("id") String id, @Context HttpServletRequest request, @Context HttpServletResponse response) throws Exception {
        String userId = id != null ? id : "";
        String data = java.net.URLDecoder.decode(userId, "UTF-8");
        response.setContentType("text/html");
        return Response.ok(data, MediaType.TEXT_HTML).build();
    }
}

File: Benchmarks/normal/python/django/testcode/benchmark_test_00257.py
# SPDX-License-Identifier: Apache-2.0
from django.http import JsonResponse
from django.utils.safestring import mark_safe
from django.http import HttpResponse
from django.template import Template, Context


def BenchmarkTest00257(request):
    user_id = request.GET.get('id', '')
    data = '{}'.format(user_id)
    return HttpResponse(Template('{{ value }}').render(Context({'value': data})))
